cmd/action/internal/uri/uri.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2021-present The Atlas Authors. All rights reserved.
+// This source code is licensed under the Apache 2.0 license found
+// in the LICENSE file in the root directory of this source tree.
+
+package uri
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+var inMemory = regexp.MustCompile("^file:.*:memory:$|:memory:|^file:.*mode=memory.*")
+
+// SqliteExists returns nil if the sqlite dsn is in memory or file exists, otherwise error.
+func SqliteExists(dsn string) error {
+	if !inMemory.MatchString(dsn) {
+		return fileExists(dsn)
+	}
+	return nil
+}
+
+func fileExists(dsn string) error {
+	s := strings.Split(dsn, "?")
+	fn := dsn
```

Comment:
fn usually means "functions". use the dsn instead
```suggestion
```

Reply:
it's a file name (potential), I'll use f